### ISSUE TITLE
Remove all the pre-split compatibility code

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3581]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#3581|
|Type|task|
|Created|2008-12-16 18:38:58Z|
|Branch|https://github.com/twisted/twisted/tree/remove-pre-split-code-3581-2|

[twisted/im.py](twisted/im.py) is one the pre-split modules that hasn't been touched in 4 years, it's probably time to get rid of it? It also contains a really old project [link](http://projects.twistedmatrix.com/words), which could be fixed instead if for some reason this file has to stay there.

<details><summary>Searchable metadata</summary>

```
trac-id__3581 3581
type__task task
reporter__thijs thijs
priority__high high
milestone__ 
branch__branches_remove_pre_split_code_3581_2 branches/remove-pre-split-code-3581-2
branch_author__thijs__exarkun thijs, exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1229452738000000 1229452738000000
changetime__1238423530000000 1238423530000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

Updating description

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [25683]) Branching to 'remove-pre-split-code-3581'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Modules that will go:

```
./im.py:util.moduleMovedForSplit('twisted.im', 'twisted.words.im',
./protocols/dns.py:util.moduleMovedForSplit('twisted.protocols.dns', 'twisted.names.dns',
./protocols/ethernet.py:util.moduleMovedForSplit('twisted.protocols.ethernet', 'twisted.pair.ethernet',
./protocols/http.py:util.moduleMovedForSplit('twisted.protocols.http', 'twisted.web.http',
./protocols/imap4.py:util.moduleMovedForSplit('twisted.protocols.imap4', 'twisted.mail.imap4',
./protocols/ip.py:util.moduleMovedForSplit('twisted.protocols.ip', 'twisted.pair.ip',
./protocols/irc.py:util.moduleMovedForSplit('twisted.protocols.irc', 'twisted.words.protocols.irc',
./protocols/jabber.py:util.moduleMovedForSplit('twisted.protocols.jabber', 'twisted.words.protocols.jabber',
./protocols/msn.py:util.moduleMovedForSplit('twisted.protocols.msn', 'twisted.words.protocols.msn',
./protocols/nntp.py:util.moduleMovedForSplit('twisted.protocols.nntp', 'twisted.news.nntp',
./protocols/oscar.py:util.moduleMovedForSplit('twisted.protocols.oscar', 'twisted.words.protocols.oscar',
./protocols/pop3.py:util.moduleMovedForSplit('twisted.protocols.pop3', 'twisted.mail.pop3',
./protocols/raw.py:util.moduleMovedForSplit('twisted.protocols.raw', 'twisted.pair.raw',
./protocols/rawudp.py:util.moduleMovedForSplit('twisted.protocols.rawudp', 'twisted.pair.rawudp',
./protocols/smtp.py:util.moduleMovedForSplit('twisted.protocols.smtp', 'twisted.mail.smtp',
./protocols/sux.py:util.moduleMovedForSplit('twisted.protocols.sux', 'twisted.web.sux',
./protocols/toc.py:util.moduleMovedForSplit('twisted.protocols.toc', 'twisted.words.protocols.toc',
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

`moduleMovedForSplit` can be deprecated too.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Removed the placeholder modules in r25686, and cleaned up the python.util module in r25687. I 'accidently' removed the C{moduleMovedForSplit} method which should've been deprecated first, so will fix that now.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

Removed note about ```DeprecationWarnings``` in t.p in r25688

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Deprecated ```moduleMovedForSplit``` in r25689, putting it up for review. I probably didn't do it the right way so looking forward to any pointers (that I'll incorporate in [#3264](https://github.com/twisted/twisted/issues/3264)).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

And I guess the [split faq](http://twistedmatrix.com/projects/core/documentation/upgrades/2.0/split.html) needs an update as well, by removing the NOTES for the t.protocols placeholders.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @thijstriemstra<br>@exarkun set status to `new`|
|-|-|

Please put all of the unrelated util.py changes into a ticket for general util.py coding standard cleanups.

Coding standard cleanups in `moduleMovedForSplit` would be fine for this branch, but making changes to every other API in util.py is too annoying.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra commented|
|-|-|

(In [26436]) Branching to 'remove-pre-split-code-3581-2'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra removed owner|
|-|-|

Created a new branch that is ready for review (r26437), basically the same as the previous branch but now without the docstring fixes.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

The deprecation warning in `moduleMovedForSplit` needs a unit test.  This will show that it is missing a stacklevel=2 argument. :)  The function should also have a docstring pointing out that it does nothing and is just for compatibility.

Also, there's a global `warnings` import, so there doesn't still need to be one in the function.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Thanks for the review, assigning it to you so you can safely fix the rest of this ticket :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_15"></a>|@exarkun commented|
|-|-|

(In [26523]) Add test for deprecation; remove nested warnings imports

refs [#3581](https://github.com/twisted/twisted/issues/3581)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_17"></a>|@mwhudson set owner to @exarkun|
|-|-|

Looks fine, please land.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_18"></a>|@exarkun set status to `closed`|
|-|-|

(In [26541]) Merge remove-pre-split-code-3581-2

Author: thijs, exarkun
Reviewer: exarkun, mwh
Fixes: [#3581](https://github.com/twisted/twisted/issues/3581)

Remove the compatibility layer added for modules which were moved to new
locations for the 2.0 split.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
